server/core/src/https/mod.rs

```diff
@@ -183,14 +183,25 @@ impl RequestExtensions for tide::Request<AppState> {
         (eventid, hv)
     }
 
+    /// Returns the remote address of the client, based on if you've got trust_x_forward_for set in config.
     fn get_remote_addr(&self) -> Option<IpAddr> {
         if self.state().trust_x_forward_for {
-            self.remote()
+            // split the socket address off if we've got one, then parse it as an `IpAddr`
+            // xff headers don't have a port, but if we're going direct you might get one
+            let res = self
+                .remote()
+                .map(|addr| addr.split(':').next().unwrap_or(addr))
```

Comment:
I'm pretty sure that discounts the existence of IPv6 addresses.

```
 kanidmd[166700]: ed6a5f2a-56be-479a-9315-82b6dcf608c4 INFO     tide-request [ 56.7µs | 100.00% ]
 kanidmd[166700]: ed6a5f2a-56be-479a-9315-82b6dcf608c4 INFO     ┝━ ｉ [info]: Request received | event_tag_id: 9 | remote_addr: "2a01:4f8:c2c:e698::1" | http.host: "accounts-test.darmstadt.ccc.de" | http.method: "POST" | path: "/v1/auth"
 kanidmd[166700]: ed6a5f2a-56be-479a-9315-82b6dcf608c4 ERROR    ┝━ 🚨 [error]: Unable to get remote addr for auth event, refusing to proceed
 kanidmd[166700]: ed6a5f2a-56be-479a-9315-82b6dcf608c4 ERROR    ┕━ 🚨 [error]: Internal error -> Response sent | event_tag_id: 4 | message: unable to validate peer address | error_type: "?" | status: 500 - Internal Server Error
```

Reply:
Looks like a job for `SocketAddr::parse_ascii` or similar.

Reply:
Yeah, I realised last night about the IPv6 thing 😞